### PR TITLE
Speedup relative pose estimation

### PR DESCRIPTION
**Pull request by ahojnnes**
This PR changes the scheduling strategy from static to dynamic, which speeds up the synchronization of threads every 10%. Further, we also avoid needless reallocation of some memory.

### CODE REVIEW

File: glomap/estimators/relpose_estimation.cc [hunk 1]
```diff
@@ -1,57 +1,64 @@
...
-      image_t idx1 = image_pair.image_id1;
-      image_t idx2 = image_pair.image_id2;
+  const size_t num_valid_pairs = valid_pair_ids.size();
+  LOG(INFO) << "Estimating relative pose for " << num_valid_pairs << " pairs";
+
+#pragma omp parallel
+  {
+    // Define variables outside parallel for loop to reuse allocations.
+    std::vector<Eigen::Vector2d> points2D_1, points2D_2;
+    std::vector<char> inliers;
```
Comment (ahojnnes) [inline glomap/estimators/relpose_estimation.cc:26]: @lpanaf Is it intentional that we don't update the inliers of the image_pair here but keep the original colmap one?

File: glomap/estimators/relpose_estimation.cc [hunk 2]
```diff
@@ -3,69 +3,69 @@
...
           points2D_1,
           points2D_2,
-          ColmapCameraToPoseLibCamera(cameras[images[idx1].camera_id]),
-          ColmapCameraToPoseLibCamera(cameras[images[idx2].camera_id]),
+          ColmapCameraToPoseLibCamera(cameras[image1.camera_id]),
+          ColmapCameraToPoseLibCamera(cameras[image2.camera_id]),
           options.ransac_options,
           options.bundle_options,
           &pose_rel_calc,
           &inliers);
```
Comment (ahojnnes) [inline glomap/estimators/relpose_estimation.cc:57]: @lpanaf Is it intentional that we don't update the inliers of the image_pair here but keep the original colmap one?

Comment (lpanaf) [reply glomap/estimators/relpose_estimation.cc:57]: We calculate the inliers in later stage, with different two-view geometry: E, F, or H

**Review by lpanaf — APPROVED** [review]

File: glomap/estimators/relpose_estimation.cc (hunk 2 above)
Comment (lpanaf) [reply glomap/estimators/relpose_estimation.cc:57]: Seems to have some formatting problem here

Comment (ahojnnes) [reply glomap/estimators/relpose_estimation.cc:58]: Thanks, I found it. Makes sense.

**Pull request merged**